ci-operator/step-registry/aws-load-balancer/pre-install-rosa/aws-load-balancer-pre-install-rosa-commands.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+AWSCRED="${CLUSTER_PROFILE_DIR}/.awscred"
+ROSA_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/ocm-token")
+# Use mounted registry-pull-credentials secret.
+# $CLUSTER_PROFILE_DIR/pull-secret cannot be used here.
+# It doesn't have the auths for the CI build registries (e.g. registry.build01.ci.openshift.org).
+CI_REGISTRY_PULL_SECRET="/var/run/albo/registry/.dockerconfigjson"
+REGION="${LEASED_RESOURCE}"
+CLUSTER_ID=$(cat "${SHARED_DIR}/cluster-id")
+# ALBO roles have to be cleaned up, otherwise ccoctl won't create secret manifests.
+# IAM roles with the prefix below are cleaned up by rosa-sts-account-roles-delete ref.
+ACCOUNT_ROLES_PREFIX=$(cat "${SHARED_DIR}/account-roles-prefix")
+#ALBO_SRC_DIR="/go/src/github.com/openshift/aws-load-balancer-operator"
+CR_DIR="/tmp/albo-credrequests"
+MANIFEST_DIR="/tmp/albo-manifests"
+
+echo "=> configuring aws"
+if [ -f "${AWSCRED}" ]; then
+  export AWS_SHARED_CREDENTIALS_FILE="${AWSCRED}"
+  export AWS_DEFAULT_REGION="${REGION}"
+else
+  echo "Did not find compatible cloud provider cluster_profile"; exit 1
+fi
+
+echo "=> logging into rosa"
+if [ ! -z "${ROSA_TOKEN}" ]; then
+  echo "Logging into staging with offline token using rosa cli $(rosa version)"
+  rosa login --env "staging" --token "${ROSA_TOKEN}" || { echo "Login failed"; exit 1; }
+else
+  echo "Cannot login! You need to specify the offline token ROSA_TOKEN!"; exit 1
+fi
+
+echo "=> getting identity provider name"
+rosa describe cluster --cluster="${CLUSTER_ID}"
+AWS_ACCOUNT_ID=$(rosa describe cluster --cluster="${CLUSTER_ID}" | grep 'AWS Account:' | tr -d ' ' | cut -d: -f2)
+OIDC_IDP_NAME=$(rosa describe cluster --cluster="${CLUSTER_ID}" | grep 'OIDC Endpoint URL:' | tr -d ' ' | cut -d: -f2- | cut -d/ -f3-)
+IDP_ARN="arn:aws:iam::${AWS_ACCOUNT_ID}:oidc-provider/${OIDC_IDP_NAME}"
+
+echo "=> preparing credential requests"
+curl --create-dirs -o ${CR_DIR}/operator.yaml https://raw.githubusercontent.com/openshift/aws-load-balancer-operator/main/hack/operator-credentials-request.yaml
```

Comment:
We could use the local clone of the repo instead of main. 
https://github.com/openshift/release/blob/28748f6c111ab3aa7c5578d9e4dfbff54404d27a/ci-operator/step-registry/aws-load-balancer/pre-install/aws-load-balancer-pre-install-commands.sh#L9

Future proof for changes in the credentials requests.

Reply:
Yeah, that's what I wanted too but the `pre-install-rosa` step is based on [the rosa client image](https://github.com/openshift/release/pull/36655/files#diff-02e284913cc11ad96e2f25d4b77d3b0592575d58b4e264643f90cc41842abcb9), not `src` one which clones the source repository.

I think I can clone the source repository and checkout to the needed ref (can be found in `$JOB_SPEC` variable), however if you have better ideas I'm very open.

Reply:
Hmm, we could pull apart the 2 workflows to be more independent. A chain/steps containing required commands (pre, post) for just spinning up rosa and a separate more specific workflow (using conventional base images to install the operator on rosa) which uses the rosa steps/chains.

Reply:
This will increase re-usability of the steps needed for rosa setup for other operators as well.

Reply:
@thejasn : thanks for the idea! I approached this a little simpler though. I modified the existing `pre-install` step to copy the credentials requests into `${SHARED_DIR}` so that the `pre-install-rosa` step can take it from there after `pre-install`. Obviously I had to add `pre-install` step to the `install-rosa` workflow but this looks logical enough.